modules/axnet/src/smoltcp_impl/mod.rs

```diff
@@ -172,6 +174,11 @@ impl InterfaceWrapper {
         let mut sockets = sockets.lock();
         iface.poll(timestamp, dev.deref_mut(), &mut sockets);
     }
+
+    #[cfg(feature = "irq")]
+    pub fn ack_interrupt(&self) {
+        unsafe { &mut *self.dev.as_mut_ptr() }.ack_interrupt();
```

Comment:
Why not lock `self.dev`?

Reply:
Because that's interrupt handler. The lock may be held by others. But handler can't wait. It's not a thread.